src/middlewared/middlewared/plugins/cache.py

```diff
@@ -1,81 +1,58 @@
-from middlewared.schema import Any, Str, Int, accepts
-from middlewared.service import Service, private
-
 from collections import namedtuple
-import time
+from collections.abc import Callable
+from time import monotonic
+from typing import Any
 
+from middlewared.service import Service
 
-class CacheService(Service):
 
+class CacheService(Service):
     class Config:
         private = True
 
     def __init__(self, *args, **kwargs):
         super(CacheService, self).__init__(*args, **kwargs)
         self.__cache = {}
-        self.kv_tuple = namedtuple('Cache', ['value', 'timeout'])
+        self.kv_tuple = namedtuple("Cache", ["value", "timeout"])
 
-    @accepts(Str('key'))
-    def has_key(self, key):
-        """
-        Check if given `key` is in cache.
-        """
+    def has_key(self, key: str):
+        """Check if given `key` is in cache."""
         return key in self.__cache
 
-    @accepts(Str('key'))
-    def get(self, key):
+    def get(self, key: str):
         """
         Get `key` from cache.
 
         Raises:
             KeyError: not found in the cache
         """
-
         if self.__cache[key].timeout > 0:
             self.get_timeout(key)
-
         return self.__cache[key].value
 
-    @accepts(Str('key'), Any('value'), Int('timeout', default=0))
-    def put(self, key, value, timeout):
-        """
-        Put `key` of `value` in the cache.
-        """
-
+    def put(self, key: str, value: Any, timeout: int = 0):
+        """Put `key` of `value` in the cache."""
         if timeout != 0:
-            timeout = time.monotonic() + timeout
+            timeout = monotonic() + timeout
+        self.__cache[key] = self.kv_tuple(value=value, timeout=timeout)
 
-        v = self.kv_tuple(value=value, timeout=timeout)
-        self.__cache[key] = v
-
-    @accepts(Str('key'))
-    def pop(self, key):
-        """
-        Removes and returns `key` from cache.
-        """
-        cache = self.__cache.pop(key, None)
-
-        if cache is not None:
-            cache = cache.value
-
-        return cache
+    def pop(self, key: str):
+        """Removes and returns `key` from cache."""
+        try:
+            return self.__cache.pop(key, None).value
+        except AttributeError:
```

Comment:
Would it be better to avoid AttributeError in first place?

Reply:
It's quicker to catch known error then check for existence 🤷 but I can go back to original way of doing it.

Reply:
fixed it